glue/core/roi.py

```diff
@@ -674,10 +814,29 @@ def contains(self, x, y):
         result = points_inside_poly(x, y, self.vx, self.vy)
         return result
 
+    # There are several possible definitions of the centre; this is easiest to calculate
+    # (and invariant under rotation?) - do not include starting vertex twice!
+    def center(self):
+        if not self.defined():
+            raise UndefinedROI
+        if self.vx[-1] == self.vx[0] and self.vy[:-1] == self.vy[0]:
+            return np.mean(self.vx[:-1]), np.mean(self.vy[:-1])
+        else:
+            return np.mean(self.vx), np.mean(self.vy)
```

Comment:
Sometimes when drawing a region one might end up with a different density of points around the edge, so I wonder if a better definition would be to use the center of the bounding box of the unrotated vertices - this will then be more meaningful once we are able to use the UI to rotate the polygon as this will likely be done by showing the bounding box of the polygon and dragging it around to rotate. This does mean that we should retain a reference to the unrotated vertices, which I think is sensible anyway.

Reply:
The bounding box would at least save the worries if a starting node might be counted with double weight. But the bounding box has the disadvantage that it is not invariant under that rotation (unless we define "bounding box" to be aligned to x, y for the unrotated polygon only, and then rotate along with it). Having the centre change with rotations would make them non-commutative, and tumble the current concept of `rotate_to` and `rotate_by`.
Keeping the unrotated vertices is certainly good point either way, though this also leaves open what to do if vertices have been added or removed in between.

Reply:
Having `center` depend on the density of vertices admittedly can be somewhat confusing, but it can be sensibly interpreted as a kind of centre of mass for the vertex collection...

Reply:
> Sometimes when drawing a region one might end up with a different density of points around the edge, so I wonder if a better definition would be to use the center of the bounding box of the unrotated vertices - this will then be more meaningful once we are able to use the UI to rotate the polygon as this will likely be done by showing the bounding box of the polygon and dragging it around to rotate. This does mean that we should retain a reference to the unrotated vertices, which I think is sensible anyway.

This is addressed by using `self.centroid` (see the "mock points" test).
Currently it is not keeping an explicit reference to the original vertices, just to the angle, which makes it possible in principle to restore them by rotating back to `theta=0`.
But I have still made the centre of rotation an optional input for now, since there may be legit reason to rotate around the mean position, bbox centre or even a corner or other point. This will produce confusing/inconsistent results when applying subsequent `rotate_by` operations around different centres though, so needs more discussion if this really is a good idea to allow.